catalyst/engines/torch.py

```diff
@@ -304,25 +316,46 @@ def get_engine(self):
 
     def __init__(
         self,
-        address: str = "localhost",
-        port: str = "12345",
-        backend: str = "nccl",
-        world_size: int = None,
+        address: str = None,
+        port: Union[str, int] = None,
+        ddp_kwargs: Dict[str, Any] = None,
+        process_group_kwargs: Dict[str, Any] = None,
     ):
         """Init."""
         super().__init__()
-        self.address = address
-        self.port = port
-        self.backend = backend
+        self.address = address or "localhost"
+        self.port = port or 12345
         self._rank = 0
-        self._world_size = world_size or torch.cuda.device_count()
         self.device = None
 
+        if ddp_kwargs is None:
+            ddp_kwargs = {}
+        self.ddp_kwargs = copy.deepcopy(ddp_kwargs)
+        if "device_ids" not in self.ddp_kwargs:
+            self.ddp_kwargs["device_ids"] = [self.device]
```

Comment:
we should move this under `def setup_process(self, rank: int = -1, world_size: int = 1):` in the end
cause `self.device = None`